typespec-extension/src/external-schemas.ts

```diff
@@ -206,7 +206,13 @@ export function getFileDetailsSchema(
   stringSchema: StringSchema,
   processSchemaFunc: (type: SdkType) => Schema,
 ): ObjectSchema {
+  let fileSdkType: SdkModelType | undefined;
   if (property.type.kind === "model") {
+    fileSdkType = property.type;
+  } else if (property.type.kind === "array" && property.type.valueType.kind === "model") {
+    fileSdkType = property.type.valueType;
+  }
```

Comment:
Bug was due to the revert back to Array<File> from TCGC property.

So here we need to either take the File from property, or the File inside the Array.